src/Compiler/Checking/CheckFormatStrings.fs

```diff
@@ -72,15 +72,29 @@ let makeFmts (context: FormatStringCheckContext) (fragRanges: range list) (fmt:
     let sourceText = context.SourceText
     let lineStartPositions = context.LineStartPositions
 
+    // Number of curly braces required to delimiter interpolation holes
+    // = Number of $ chars starting a (triple quoted) string literal
+    // Set when we process first fragment range, default = 1
+    let mutable delimLen = 1
+
     let mutable nQuotes = 1
     [ for i, r in List.indexed fragRanges do
         if r.StartLine - 1 < lineStartPositions.Length && r.EndLine - 1 < lineStartPositions.Length then
             let startIndex = lineStartPositions[r.StartLine - 1] + r.StartColumn
             let rLength = lineStartPositions[r.EndLine - 1] + r.EndColumn - startIndex
             let offset =
                 if i = 0 then
-                    match sourceText.GetSubTextString(startIndex, rLength) with
-                    | PrefixedBy "$\"\"\"" len
+                    let fullRangeText = sourceText.GetSubTextString(startIndex, rLength)
```

Comment:
Would be great if we could scan the sourceText at appropriate positions without the intermmediate allocation.
But I know that are no nice idioms for it if we cannot use Span<> :((

Reply:
After discussing pros and cons offline, I'll leave this as is